### DISCUSSION TITLE
How to change text of specific line in multiline textbox in WPF?

### DISCUSSION THREAD

**Discussion by VahidEra** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3026] [Q&A] [answered]
Hi guys,
I have a multi-line text box in my WPF project, and I want to change the specified line text. I used to do this easily in Win form, but there is no Lines option in WPF Text box. For example, I used to do this in Windows Form for change text of specific line like this:

            Dim lines As String() = VTextBox1.Lines

            lines(Specific line number) = "This is a test."

            VTextBox1.Lines = lines

But this piece of cod, not working in WPF textbox.

> Do I have any chance that I do this in the WPF textbox?
I have no idea how to do that.

**Comment by nicolaihenriksen** [comment, marked as answer]
@VahidEra This question does not seem related to MaterialDesignInXamlToolkit in any way, but a general WPF question? I think there are probably other more appropriate forums for such a question.

With that said, I think what you are asking is somewhat of a special case which is difficult to provide a generic solution for. A `TextBox` can be "multiline" in various ways:

- You can set `AcceptsReturn=True` (which is probably your scenario) where each line is separated by a `Environment.NewLine` string.
- You can set the `TextWrapping` property to one of the "wrapping" options.
- You can have a combination of the two above.

For the first scenario, it could be straight forward by simply pulling out the `TextBox.Text` property, splitting it on `Environtment.NewLine`, modifying the string at the desired index, and the joining them back together again. However for the latter 2, you do not have that option and you are stuck having to basically calculate "the lines" based on the controls width along with the typeface used; this gets ugly real quick!

**Reply by VahidEra** [reply]
@nicolaihenriksen Dear Nicolai,
I am using MaterialDesignInXamlToolkit in WPF and I thought with myself Perhaps it would be to do this For the same Essen I asked the question here, but in any case, thank you so much for help.

**Reply by VahidEra** [reply]
@nicolaihenriksen According to your explanation, I solved this issue, and it worked perfectly for me.

`
Dim lines As String() = VTextBox1.Text.Split(vbLf)
`
`
Dim lineNumber As Integer = 5
`
`
If lines.Length > lineNumber Then
`
`
      lines(lineNumber) = "This is a test."
`
`
VTextBox1.Text = String.Join(vbLf, lines)
`
`
end if
`
